### PR TITLE
feat/start and stop

### PR DESCRIPTION
**Pull request by diasdavid**
Needed to finish https://github.com/ipfs/js-ipfs/pull/644

### CODE REVIEW

**Comment by diasdavid** [comment]
@gavinmcdermott could you accept the invitation for the libp2p org? 
image

File: src/index.js [hunk 1]
```diff
@@ -200,6 +192,62 @@ class FloodSub extends EventEmitter {
...
 
   /**
+   * Mounts the floodsub protocol onto the libp2p node and sends our
+   * subscriptions to every peer conneceted
+   *
+   * @param {Function} callback
+   * @returns {undefined}
+   *
+   */
+  start (callback) {
```
Comment (dignifiedquire) [inline src/index.js]: remembering the discussions for the better js-ipfs start and stop interface, could we make this callback just bind to a start event and similarly for stop?

File: src/index.js [hunk 2]
```diff
@@ -200,6 +192,62 @@ class FloodSub extends EventEmitter {
...
+    // Dial already connected peers
+    const peerInfos = values(this.libp2p.peerBook.getAll())
+
+    peerInfos.forEach((peerInfo) => {
+      this._dialPeer(peerInfo)
+    })
+
+    setImmediate(() => {
+      this.started = true
+      callback()
```
Comment (dignifiedquire) [inline src/index.js]: why a callback at all if this is actually snyc?

File: src/index.js (hunk 1 above)
Comment (diasdavid) [inline src/index.js]: I believe the improving init -- https://github.com/ipfs/js-ipfs/issues/556 -- discussion comes from the annoyance of having 3 steps to start js-ipfs (init, load, goOnline), while in bitswap, floodsub, libp2p it is really just one

File: src/index.js (hunk 2 above)
Comment (diasdavid) [inline src/index.js]: mostly future proofing, but I was also asking myself if we should wait for the dials to happen.

**Comment by gavinmcdermott** [comment]
@diasdavid - thanks for the ping—just accepted. Even though I've been a bit IDEO-work-focused this week, I've been tracking your guys' progress via the latest PRs—beautiful! I'm almost wrapped up getting the latest version of Nomad ready for a 2 week intensive build/exploration. In a nutshell, by this weekend:
- ported fully over to JS-land (off of the go daemon)
- using this initial version of pubsub
- lots of tests!

We'll get a bunch of mileage out of the latest updates next week (meaning pubsub should be getting hammered on)...so I'm sure useful new questions will begin to surface. Look forward to being back in the loop and talk soon!

**Pull request merged**